lms/djangoapps/lti_provider/tasks.py

```diff
@@ -72,6 +72,54 @@ def increment_assignment_versions(course_key, usage_key, user_id):
     return assignments
 
 
+class CourseWithoutVersionsAndBranches(object):
+
+    def __init__(self, course):
+        self.course = course
+        self._store = {'location': {}, 'parent': {}}
+
+    def __enter__(self):
+        if settings.CELERY_ALWAYS_EAGER:
+            self._remove_versions_and_branches(self.course)
+        return self.course
+
+    def _remove_versions_and_branches(self, course):
+        for child in course.get_children():
+            location_version = child.location.version
+            location_branch = child.location.branch
+            if location_version or location_branch:
+                child.location = child.location.replace(version=None, branch=None)
+                self._store['location'][child.location] = {'version': location_version, 'branch': location_branch}
+
+            parent_version = child.parent.version
+            parent_branch = child.parent.branch
+            if parent_version or parent_branch:
+                child.parent = child.parent.replace(version=None, branch=None)
+                self._store['parent'][child.parent] = {'version': parent_version, 'branch': parent_branch}
+
+            self._remove_versions_and_branches(child)
+
+    def __exit__(self, type, value, traceback):
+        if settings.CELERY_ALWAYS_EAGER:
+            self._restore_versions_and_branches(self.course)
+
+    def _restore_versions_and_branches(self, course):
```

Comment:
@douglashall I'm not sure. Is it a bug that course nodes contain branch and version info? Because it looks like it is regular situation and it should be. I don't know so I decided to restore all this information on exit. `send_composite_outcome` works when the new `grade_event` is fired. So may be there are some other functions that are listen this event.